aten/src/ATen/native/mps/operations/Indexing.mm

```diff
@@ -295,17 +303,14 @@ Tensor nonzero_fallback(const Tensor& self) {
           MPSGraph* mpsGraph = make_mps_graph();
           newCachedGraph = new CachedGraph(mpsGraph);
 
-          MPSGraphTensor *inputTensor = mpsGraphRankedPlaceHolder(mpsGraph, getMPSScalarType(self.scalar_type()), apparentInputShape);
+          MPSGraphTensor *inputTensor = mpsGraphUnrankedPlaceHolder(mpsGraph, getMPSScalarType(self.scalar_type()));
```

Comment:
I guess it makes sense to make every op unranked and just put the dims in the key.